utils/upload_everything/upload_everything.py

```diff
@@ -67,41 +67,115 @@
+            while True:
+                start_time, end_time = time_range
+                concatenated_string = f"{user} {camera_name} {start_time} to {end_time} all tag:box"
+                call_time = datetime.now()
+
+                # Get the queue lengths for the specified boxes, will be [] if none specified
+                queue_lengths = get_upload_queue_lengths(user=user, token=token, hostname=hostname,
+                                                         device_ids_to_check=device_ids_to_check, dry_run=dry_run)
+                queue_lengths_string = ' '.join(str(length) for length in queue_lengths)
+                queue_is_full = not all(value < upload_queue_threshold for value in queue_lengths)
+
+                # Don't request tag:box upload for dry runs or if the queue is filling up
+                if not dry_run and not queue_is_full:
+                    response = make_api_request(concatenated_string, token, hostname)
+                    search_url = response.url.replace("/api/search?text=", "/app/#search;q=").replace("+tag%3Abox", "")
+                    upload_commands_count = 0
+                    uploading_devices = ''
+                    is_working = False
+                    if response.status_code // 100 == 2:  # in the 2xx range
+                        response_data = response.json()
+                        if 'operations' in response_data and 'upload_commands' in response_data[
+                            'operations'] and isinstance(response_data['operations']['upload_commands'], list):
+                            upload_commands = response_data['operations']['upload_commands']
+                            upload_commands_count = len(upload_commands)
+                            uploading_devices = ' '.join(
+                                upload_command["device_id_internal"] for upload_command in upload_commands)
+                            is_working = True
+                    else:
+                        sys.stderr.write(f"Error making upload request: {response.status_code} ({response.reason})\n")
+
+                    print(
+                        f"{call_time.isoformat()},{response.url},{response.status_code},{upload_commands_count},{queue_lengths_string},{uploading_devices},{search_url}")
+                    sys.stdout.flush()
+                    if is_working:
+                        time.sleep(wait_seconds)
+                    break
+
+                elif not dry_run and queue_is_full:
+                    # One or more of the queues are full
+                    current_time = datetime.now()
+                    elapsed_time = current_time - iteration_start_time
+                    elapsed_total_seconds = elapsed_time.total_seconds()
+
+                    # Break iteration if reached max wait time
+                    if elapsed_total_seconds > max_wait_time:
+                        sys.stderr.write(f"Waited for max {max_wait_time}s for user {user}, camera {camera_name}\n")
```

Comment:
include `{start_time} to {end_time}` so that we can easily track which time periods we have to retry.

Reply:
It's included in the `output.csv`, sample:

```
2024-04-30T13:36:53.695004,gabriella@camiolog.com Liverpool-Test2 11 7pm PT April 28th to 8pm PT April 28th all tag:box,N/A,N/A,1,N/A,N/A
```

but I will add it to the log as well.